ParseLiveQuery/src/test/java/com/parse/TestParseLiveQueryClient.java

```diff
@@ -345,6 +347,28 @@ public void testEmptySessionTokenOnSubscribe() {
                 contains("\"sessionToken\":\"the token\"")));
     }
 
+    @Test
+    public void testDisconnectOnBackgroundThread() throws Exception {
+        PauseableExecutor executor = new PauseableExecutor();
+        // Recreate the client with a PauseableExecutor
+        parseLiveQueryClient = ParseLiveQueryClient.Factory.getClient(new URI(""), new WebSocketClientFactory() {
+            @Override
+            public WebSocketClient createInstance(WebSocketClient.WebSocketClientCallback webSocketClientCallback, URI hostUrl) {
+                TestParseLiveQueryClient.this.webSocketClientCallback = webSocketClientCallback;
+                webSocketClient = mock(WebSocketClient.class);
+                return webSocketClient;
+            }
+        }, executor);
+
+        reconnect();
+        executor.pause();
```

Comment:
The intent of pause/unpause (including their names) is inspired by the Robolectric scheduler mocks, which allow pausing (must manually advance), unpausing (runs everything blocking), and advancing a paused scheduler one task at a time.

In this case, the `TaskQueueTestHelper` would not work for us here, because the LiveQuery client expects an `Executor`, not a TaskQueue.

And the TaskQueueTestHelper actually is pausing and unpausing the TaskQueue (the javadoc even explicitly says "pauses the TaskQueue") -- so I'm not sure that I agree with the naming of that class's methods.  "enqueue" and "dequeue" does not imply in any way that it will block or pause the TaskQueue:  "Enqueue" means that it is adding a task to the queue, nothing else.

If the ParseLiveQueryClientImpl were actually using a TaskQueue, then we could make use of that class;  but it doesn't use TaskQueue at all.

I think for the purposes of a text fixture, it really doesn't matter what it's called.  My vote would be for pause and unpause… but I'm happy to rename the method if you think that would be a point of confusion?